components/automate-ui/src/app/modules/infra-proxy/chef-server-details/chef-server-details.component.spec.ts

```diff
@@ -0,0 +1,136 @@
+        MockComponent({ selector: 'chef-icon' }),
+        MockComponent({ selector: 'chef-loading-spinner' }),
+        MockComponent({ selector: 'chef-option' }),
+        MockComponent({ selector: 'chef-page-header' }),
+        MockComponent({ selector: 'chef-subheading' }),
+        MockComponent({ selector: 'chef-toolbar' }),
+        MockComponent({ selector: 'chef-table-new' }),
+        MockComponent({ selector: 'chef-table-header' }),
+        MockComponent({ selector: 'chef-table-body' }),
+        MockComponent({ selector: 'chef-table-row' }),
+        MockComponent({ selector: 'chef-table-header-cell' }),
+        MockComponent({ selector: 'chef-table-cell' }),
+        MockComponent({ selector: 'a', inputs: ['routerLink'] }),
+        ChefServerDetailsComponent
+      ],
+      providers: [
+        FeatureFlagsService
+      ],
+      imports: [
+        FormsModule,
+        ReactiveFormsModule,
+        RouterTestingModule,
+        StoreModule.forRoot(ngrxReducers, { runtimeChecks })
+      ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ChefServerDetailsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('create org', () => {
+    let store: Store<NgrxStateAtom>;
+    const org = <Org> {
+        id: '1',
+        name: 'new org',
+        admin_user: 'new org user',
+        admin_key: 'new admin key'
+      };
+
+    beforeEach(() => {
+      store = TestBed.get(Store);
+    });
+
+    it('openCreateModal opens modal', () => {
+      expect(component.createModalVisible).toBe(false);
+      component.openCreateModal('create');
+      expect(component.createModalVisible).toBe(true);
+    });
+
+    it('opening create modal resets name, description, fqdn and ip_address to empty string', () => {
+      component.openCreateModal('create');
+      expect(component.orgForm.controls['name'].value).toBe('');
+      expect(component.orgForm.controls['admin_user'].value).toBe('');
+      expect(component.orgForm.controls['admin_key'].value).toBe('');
+    });
+
+    it('on conflict error, modal is open with conflict error', () => {
+      spyOn(component.conflictErrorEvent, 'emit');
+      component.openCreateModal('create');
+      component.orgForm.controls['name'].setValue(org.name);
+      component.orgForm.controls['admin_user'].setValue(org.admin_user);
+      component.orgForm.controls['admin_key'].setValue(org.admin_key);
+      component.createServerOrg();
+
+      const conflict = <HttpErrorResponse>{
+        status: HttpStatus.CONFLICT,
+        ok: false
+      };
+      store.dispatch(new CreateOrgFailure(conflict));
+
+      expect(component.createModalVisible).toBe(true);
+    });
+
+    it('on create error, modal is closed with failure banner', () => {
```

Comment:
Technically this test name is wrong because you are not ensuring that there is a failure banner. It would be nice if you can, but if it is too much work, this would be sufficient:
```suggestion
    it('on create error, modal is closed (because error is handled by failure banner)', () => {
```